agent/eni/watcher/watcher_linux.go

```diff
@@ -147,16 +147,26 @@ func (udevWatcher *UdevWatcher) reconcileOnce() error {
 // sendENIStateChange handles the eni event from udev or reconcile phase
 func (udevWatcher *UdevWatcher) sendENIStateChange(mac string) {
 	eniAttachment, ok := udevWatcher.shouldSendENIStateChange(mac)
-	if ok {
-		go func(eni *api.ENIAttachment) {
-			eni.Status = api.ENIAttached
-			log.Infof("Emitting ENI change event for: %v", eni)
-			udevWatcher.eniChangeEvent <- api.TaskStateChange{
-				TaskARN:    eni.TaskARN,
-				Attachment: eni,
-			}
-		}(eniAttachment)
+	if !ok {
+		if eniAttachment != nil {
```

Comment:
This will also remove the eni attachment that status has been sent, and still managed by agent. We should exclude that whether add a check here or return nil in shouldSendENIStateChange. Also please add this case in your test.

Reply:
Isn't that a good thing? I don't think we need to persist the attachment if we've already acknowledged with the backend that Agent has seen the ENI. The only weirdness here is that Attachment data from the state will be cleaned up before the task data, which I still think is not a problem.

Reply:
Remove it from the agent state while it's still managed by ECS sounds not right to me. One rare case I can think of is if customer can know the number of enis managed by ecs by check the eniattachment field in the state file.

Reply:
I'd argue that you should be able to get that `DescribeContainerInstance` API. But, I agree that it may lead to a weird behavior. I'll exclude `IsSent()` from this.